src/pds/ingress/pds-nucleus-datasync-completion-trigger.py

```diff
@@ -0,0 +1,70 @@
+"""
+============================================================
+pds-nucleus-datasync-completion-trigger.py
+============================================================    =
+
+Lambda function that get triggered when the PDS Nucleus Staging S3 Bucket to EFS datasync task is completed.
+This function reads the list of data transfer verification reports created by a specific DataSync task
+and asynchronously call the pds-nucleus-datasync-completion.py for each data transfer verification
+report JSON file.
+
+"""
+
+import boto3
+import logging
+import json
+import os
+
+datasync_reports_s3_bucket_name = "pds-nucleus-datassync-reports"
+
+logger = logging.getLogger("pds-nucleus-datasync-completion-trigger")
+logger.setLevel(logging.DEBUG)
+logger.addHandler(logging.StreamHandler())
+
+s3 = boto3.resource('s3')
+s3_client = boto3.client('s3')
+
+s3_bucket_name = "pds-nucleus-staging"
+db_clust_arn = os.environ.get('DB_CLUSTER_ARN')
+db_secret_arn = os.environ.get('DB_SECRET_ARN')
+efs_mount_path = os.environ.get('EFS_MOUNT_PATH')
+
+rds_data = boto3.client('rds-data')
+
+# Define the client to interact with AWS Lambda
+client = boto3.client('lambda')
+
+
+def lambda_handler(event, context):
+    """ Lambda Handler """
+
+    logger.info(f"Lambda Request ID: {context.aws_request_id}")
+    logger.info(f"Event: {event}")
+    resource = event['resources']
+
+    resource_list = resource[0].split("/")
+    task_id = resource_list[1]
+    exec_id = resource_list[3]
+
+    prefix = f"Detailed-Reports/{task_id}/{exec_id}/{exec_id}.files-verified-"
+
+    datasync_reports_s3_bucket = s3.Bucket(datasync_reports_s3_bucket_name)
+
+    list_of_files = []
+
+    # Loop through the list of json files with the prefix files-verified-
+    for transfer_report in datasync_reports_s3_bucket.objects.filter(Prefix=prefix):
+        # Define the input parameters that will be passed
+        # on to the child function
+        inputParams = {
+            "s3_key": transfer_report.key
+        }
+
+        logger.debug(f"inputParams: {str(inputParams)}")
```

Comment:
Technically, this should be `logger.debug("inputParams: %s", inputParams)` since if "debug" is off then the string interpolation is skipped.

But the `DEBUG` level is hard-coded at the top of the file, so it's okay.

Reply:
@nutjob4life , we discussed about % vs `modern f-strings` 😄  in a previous pull request and kind of concluded that it is a personal preference. 
https://realpython.com/python-f-strings/

Reply:
Yes, but: https://realpython.com/python-f-strings/#lazy-evaluation-in-logging
```python
>>> import logging, timeit
>>> logging.basicConfig(level=logging.WARNING)
>>> timeit.timeit('import logging; value = 42; logging.debug(f"hello {value}")', number=9999999)
2.9089915420045145
>>> timeit.timeit('import logging; value = 42; logging.debug("hello %d", value)', number=9999999)
2.5685420420049923
```

Reply:
yes, but https://realpython.com/python-f-strings/#key-takeaways :-)